### PR TITLE
wrote unit tests for instructorassignmyTA and instructorgetinstrTAAsnmt

### PR DESCRIPTION
**Pull request by Randall-S4**
might need to edit the models file because they currently crash

### CODE REVIEW

File: TAScheduler/UnitTests/tests.py [hunk 1]
```diff
@@ -2898,3 +2898,90 @@ def test_none_to_remove(self):
...
+        lab = Lab.objects.create(section=self.section)
+        self.assertIsNone(lab.ta, "TA is already assigned to this section")
+
+        self.instructor.assignMyTA(self.ta, self.section)
+
+        # Postconditions
+        lab.refresh_from_db()  # Refresh to get the latest data
+        self.assertEqual(lab.ta, self.ta, "TA was not correctly assigned to the section")
+
+    def test_assign_ta_not_in_instructor_course(self):
```
Comment (AlecBrookens) [inline TAScheduler/UnitTests/tests.py:2939]: Are you sure there aren't other things that could fail, which we need to test for successful failure? Are there more negative cases you can check?

**Review by AlecBrookens — APPROVED** [review]
Good stuff, maybe a few more negative cases.

File: TAScheduler/UnitTests/tests.py [hunk 2]
```diff
@@ -2898,3 +2898,90 @@ def test_none_to_remove(self):
...
+        self.instructor = Instructor.objects.create(user=self.instructor_user, max_assignments=3)
+
+        # Creating course and sections
+        self.course = Course.objects.create(course_id=123, semester="Fall", name="Sample Course",
+                                            description="A sample course", num_of_sections=2, modality="Online")
+        self.section = Section.objects.create(section_id=456, course=self.course, location="Room 101",
+                                              meeting_time="2023-09-01 09:00")
+
+        # Assign instructor to course
+        InstructorToCourse.objects.create(instructor=self.instructor, course=self.course)
```
Comment (Kiran-Ee) [inline TAScheduler/UnitTests/tests.py:2925]: need to do TAToCourse() 

Add a test for "assigning TA not assigned to me course": 1 for the TA saved to another course and 1 for the TA not saved to any courses

File: TAScheduler/UnitTests/tests.py [hunk 3]
```diff
@@ -2898,3 +2898,90 @@ def test_none_to_remove(self):
...
+
+    def test_assign_ta_to_section(self):
+        # Preconditions
+        TAToCourse.objects.create(ta=self.ta, course=self.course)  # This ensures the TA is linked to the course.
+        lab = Lab.objects.create(section=self.section)
+        self.assertIsNone(lab.ta, "TA is already assigned to this section")
```
Comment (Kiran-Ee) [inline TAScheduler/UnitTests/tests.py:2931]: I don't think this is necessary, I also think that line 2930 will raise an exception (or an error) because of the way the model is stepup.

File: TAScheduler/UnitTests/tests.py [hunk 4]
```diff
@@ -2898,3 +2898,90 @@ def test_none_to_remove(self):
...
+
+    def test_assign_ta_not_in_instructor_course(self):
+        # Create a TA that is not in the instructor's course
+        other_user = User.objects.create(email_address="other_ta@example.com", password="password789",
+                                         first_name="OtherTAFirstName", last_name="OtherTALastName",
+                                         home_address="Other TA Address", phone_number=1122334455)
+        other_ta = TA.objects.create(user=other_user, grader_status=True, max_assignments=3)
+
+        with self.assertRaises(ValueError):
+            self.instructor.assignMyTA(other_ta, self.section)
```
Comment (Kiran-Ee) [inline TAScheduler/UnitTests/tests.py:2947]: It might be better to make sure the section you're trying to assignt the TA to is not full so the error to be thrown is coming from "can't assign ta not in your course" rather than "can't assign ta to full section"

File: TAScheduler/UnitTests/tests.py [hunk 5]
```diff
@@ -2898,3 +2898,90 @@ def test_none_to_remove(self):
...
+        # Get TA assignments
+        assignments = self.instructor.getInstrTAAsgmt()
+
+        # Check the content of the assignments list
+        expected_assignment = {
+            "taID": self.ta.id,
+            "secID": self.section.id,
+            "courseID": self.course.id
+        }
+        self.assertIn(expected_assignment, assignments, "TA assignment not found in instructor assignments")
```
Comment (Kiran-Ee) [inline TAScheduler/UnitTests/tests.py:2987]: add a test for assigning a TA to another course; ensure that this method does NOT return that TA.

**Review by Kiran-Ee — COMMENTED** [review]
Slighty adjustments and additional tests.

**Review by KevinMonahanUWM — CHANGES_REQUESTED** [review]
Please look over my comments. I might be wrong but unless something huge has changed you should not be calling these method with a database object

**Review by Kiran-Ee — APPROVED** [review]
nice work - thanks for making the changes!

File: TAScheduler/UnitTests/tests.py (hunk 2 above)
Comment (JosiahDeYong) [reply TAScheduler/UnitTests/tests.py:2925]: I think that you don't assign the courses in setup

**Review by JosiahDeYong — APPROVED** [review]
I don't see any problems maybe just add more tests if you can think of any
